Fix log file location in acceptance

tbg opened the pull request:
with `-l .` and in `TestX`, the log dir should be `./TestX` but it had been
shifted by introduction of additional members in the call stack.
Added an assertion to prevent this in the future.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4852)

<!-- Reviewable:end -->

tamird commented on acceptance/util_test.go:
```diff
@@ -141,9 +142,11 @@ func StartCluster(t *testing.T, cfg cluster.TestConfig) (c cluster.Cluster) {
-			if _, _, fun := caller.Lookup(1); fun != "" {
-				logDir = filepath.Join(logDir, fun)
+			_, _, fun := caller.Lookup(3)
+			if ok, _ := regexp.MatchString("^(Test|Benchmark)", fun); !ok {
```
nit: use regexp.MustCompile

tamird commented:
LGTM

tbg committed:
Fix log file location in acceptance

with `-l .` and in `TestX`, the log dir should be `./TestX` but it had been
shifted by introduction of additional members in the call stack.
Added an assertion to prevent this in the future.
